writeengine/wrapper/we_colop.cpp

```diff
@@ -507,14 +517,27 @@ int ColumnOp::allocRowId(const TxnID& txnid, bool useStartingExtent,
                 {
                     rc = readBlock(newCol.dataFile.pFile, buf, newHwm);
 
-                    if ( rc != NO_ERROR)
-                        return rc;
+                    // MCOL-498 add a block.
+                    // DRRTUY TODO Given there is no more hwm pre-allocated we
+                    // could extend the file once to accomodate all records.
+                    if (rc != NO_ERROR)
+                    {
+                        if (rc == ERR_FILE_EOF)
+                        {
+                            uint64_t emptyVal = getEmptyRowValue(newCol.colDataType, newCol.colWidth);
+                            setEmptyBuf(buf, BYTE_PER_BLOCK, emptyVal, newCol.colWidth);
+                            RETURN_ON_ERROR(saveBlock(newCol.dataFile.pFile, buf, newHwm));
```

Comment:
Am I correct in believing that these calls to saveBlock() replace the parts of write_row() deleted below?

Reply:
Precisely.